src/overlays/actors/ovl_En_Tg/z_en_tg.c

```diff
@@ -76,37 +91,193 @@ static DamageTable D_809901F8 = {
+    }
+}
+
+void EnTg_Update(Actor* thisx, PlayState* play) {
+    EnTg* this = THIS;
+
+    this->actionFunc(this, play);
+    Actor_UpdateBgCheckInfo(play, &this->actor, 0.0f, 0.0f, 0.0f, 4U);
+    EnTg_UpdateSkelAnime(this, play);
+    EnTg_UpdateHearts(play, this->enTgHeartEffect, ARRAY_COUNT(this->enTgHeartEffect));
+    EnTg_UpdateCollider(this, play);
+}
+
+s32 EnTg_OverrideLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    return 0;
+}
+
+void EnTg_PostLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    EnTg* this = THIS;
+    Vec3f zeroVec = { 0.0f, 0.0f, 0.0f };
+
+    if (limbIndex == HONEY_AND_DARLING_LIMB_MAN_HEAD) {
+        Matrix_MultVec3f(&zeroVec, &this->actor.focus.pos);
+    }
+}
+
+void EnTg_Draw(Actor* thisx, PlayState* play) {
+    EnTg* this = THIS;
+    GraphicsContext* gfxCtx;
+
+    Matrix_Push();
+    EnTg_DrawHeart(play, this->enTgHeartEffect, ARRAY_COUNT(this->enTgHeartEffect));
+    Matrix_Pop();
+
+    OPEN_DISPS(play->state.gfxCtx);
+    func_8012C28C(play->state.gfxCtx);
+
+    gDPPipeSync(POLY_OPA_DISP++);
+    gSPSegment(POLY_OPA_DISP++, 0x08, Gfx_EnvColor(play->state.gfxCtx, 0, 50, 160, 0));
+    gSPSegment(POLY_OPA_DISP++, 0x09, Gfx_EnvColor(play->state.gfxCtx, 255, 255, 255, 0));
+
+    SkelAnime_DrawFlexOpa(play, this->skelAnime.skeleton, this->skelAnime.jointTable, this->skelAnime.dListCount,
+                          EnTg_OverrideLimbDraw, EnTg_PostLimbDraw, &this->actor);
+
+    CLOSE_DISPS(play->state.gfxCtx);
+}
+
+/**
+ * Spawns a heart at the first enTgHeartEffect array index that's not enabled.
+ * Because of the frame counts, only two hearts are ever spawned at a time.
+ */
+void EnTg_SpawnHeart(EnTg* this, EnTgHeartEffect* enTgHeartEffect, Vec3f* heartStartPos, s32 len) {
+    Vec3f heartVelocityVec = { 0.0f, 1.5f, 0.0f };
+    Vec3f zeroVec = { 0.0f, 0.0f, 0.0f };
+    s32 i;
+
+    for (i = 0; i < len && enTgHeartEffect->isEnabled; i++, enTgHeartEffect++) {}
+
+    if (i < len) {
+        enTgHeartEffect->isEnabled = true;
+        enTgHeartEffect->pos = *heartStartPos;
+        enTgHeartEffect->velocity = heartVelocityVec;
+        enTgHeartEffect->unusedZeroVec20 = zeroVec;
+        enTgHeartEffect->scale = 0.01f;
+        enTgHeartEffect->pos.x += 4.0f * Math_SinS(this->actor.shape.rot.y);
+        enTgHeartEffect->pos.z += 4.0f * Math_CosS(this->actor.shape.rot.y);
+        enTgHeartEffect->timer = 16;
+    }
+}
+
+/**
+ * The heart path is curvy as it floats up because of the use of Math_SinS and Math_CosS.
+ */
+void EnTg_UpdateHearts(PlayState* play, EnTgHeartEffect* enTgHeartEffect, s32 len) {
+    Vec3f zeroVec = { 0.0f, 0.0f, 0.0f };
+    s16 yaw = Camera_GetInputDirYaw(GET_ACTIVE_CAM(play));
+    s32 i;
+
+    for (i = 0; i < len; i++, enTgHeartEffect++) {
+        if (enTgHeartEffect->isEnabled == 1) {
```

Comment:
Does this specifically require you to check for 1? e.g., does `if (enTgHeartEffect->isEnabled)` not work by itself? If that's the case, then:
```suggestion
        if (enTgHeartEffect->isEnabled == true) {
```